### ISSUE TITLE
Schema\Table should allow to add objects

### ISSUE THREAD

**Issue by mvorisek** [doctrine/dbal#5516]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no

#### Summary

Currently adding objects is possible with the constructor, but not later with the `Schema\Table::addXxx` methods.

This is a feature request to support usecase like:

```
$table = new \Doctrine\DBAL\Schema\Table(...);
$fk = new \Doctrine\DBAL\Schema\ForeignKeyConstraint(...);
$table->addForeignKeyConstraint($fk);
```

**Comment by morozov** [comment]
The requested method already exists: https://github.com/doctrine/dbal/blob/39a05f63ffb95baf13716a6c11e268a720bf2552/src/Schema/Table.php#L407-L413

What is the problem with using the existing methods?

**Comment by mvorisek** [comment]
It does, but it does not accept a `ForeignKeyConstraint` object as shown in the code example.

**Comment by morozov** [comment]
I agree in general. It would be consistent with the changes made in https://github.com/doctrine/dbal/pull/4949 and https://github.com/doctrine/dbal/pull/4950 and allow for better extensibility.

How do you propose to implement this?

**Comment by mvorisek** [comment]
> How do you propose to implement this?

`\Doctrine\DBAL\Schema\Table::addForeignKeyConstraint(ForeignKeyConstraint $fk)` method prototype would be the most logical, but it will impose BC break so it is more question to you guys who maintain this library.

**Comment by morozov** [comment]
I wouldn’t change anything since I don’t see a reasonable upgrade path and don’t understand all the use cases that we need to support. So unless you want to work on designing and implementing your proposal, we can close this issue.

**Comment by mvorisek** [comment]
The usecase should be allowed as this is supported by the constructor, but not later. The support can be added by a introducing a new method or supporting object as the 1st param together with the current method signature.

**Comment by morozov** [comment]
The source of confusion is that the `Table` class represents both the value object and its builder. If we look at it as a value object, then it should accept all its components via the constructor and not allow modification. If we look at it as a builder, it should provide a convenient API for building another value object from the current one. The current API addresses both of these concerns.

A good way to have a non-bloated API for both concerns is to separate the value and the builder classes. A table could create a builder and a builder could create a table. For instance:
```php
class Table {
    function edit(): TableBuilder;
}

class TableBuilder {
    function build(): Table;
}
```

Otherwise, we'll end up having two methods per component (e.g. `addForeignKey(string $table, array $columns)` and `addForeignKeyObject(ForeignKey $foreignKey)`).
